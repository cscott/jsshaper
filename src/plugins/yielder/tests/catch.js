var log = (typeof console !== "undefined") && console.log || print;
(typeof require !== 'undefined') && require('setup.js');

// test block scoping of catch

function foo() {
  var a = [];
  for (var i=0; i<10; i++) {
    try {
      throw i;
    } catch (j) {
      a[i] = function() { return j++; };
    }
  }
  log(a[0]());
  log(a[0]());
  log(a[0]());
  log(a[1]());
  log(a[1]());
}

function bar() {
  var a = [];
  for (var i=0; i<10; i++) {
    try {
      throw i;
    } catch (j) {
      a[i] = function() { return j++; };
    }
  }
  log(a[0]());
  log(a[0]());
  log(a[0]());
  log(a[1]());
  log(a[1]());
  yield;
}

foo();
log('-vs-');
bar().next();
