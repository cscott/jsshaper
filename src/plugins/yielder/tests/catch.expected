0
1
2
1
2
-vs-
0
1
2
1
2
